File: LeetCode-October-Challenge/BreakaPalindrome.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    string breakPalindrome(string palindrome) {
        int n = palindrome.size();
        if(n <= 1) return "";
        for(int i = 0; i < n / 2; ++i){
            if(palindrome[i] != 'a'){
                palindrome[i] = 'a';
                return palindrome;
            }
        }
        palindrome[n - 1] = 'b';
        return palindrome;
    }
};
int main(){
    string palindrome = "abccba";
    Solution s;
    cout<<s.breakPalindrome(palindrome)<<endl;
    return 0;
}

File: LeetCode-October-Challenge/BreakaPalindrome.java
class Solution {
    public String breakPalindrome(String palindrome) {
        int n = palindrome.length();
        char[] res = palindrome.toCharArray();
        if(n <= 1) return "";
        for(int i = 0; i < n / 2; ++i){
            if(res[i] != 'a'){
                res[i] = 'a';
                return String.valueOf(res);
            }
        }
        res[n - 1] = 'a';
        return String.valueOf(res);
    }
}
public class BreakaPalindrome {
    public static void main(String[] args) {
        String palindrome = "abccba";
        Solution s = new Solution();
        System.out.println(s.breakPalindrome(palindrome));
    }
}
